 @Test(timeout = 4000)
  public void test0()  throws Throwable  {
      ExceptionMapper exceptionMapper0 = ExceptionMapper.getInstance();
      Class<Exception> class0 = Exception.class;
      ExceptionHandlerImpl<Exception> exceptionHandlerImpl0 = (ExceptionHandlerImpl<Exception>) mock(ExceptionHandlerImpl.class, new ViolatedAssumptionAnswer());
      doReturn((String) null, (String) null).when(exceptionHandlerImpl0).toString();
      exceptionMapper0.map(class0, exceptionHandlerImpl0);
      MockException mockException0 = new MockException();
      ExceptionHandlerImpl exceptionHandlerImpl1 = exceptionMapper0.getHandler((Exception) mockException0);
      ;
}
 @Test(timeout = 4000)
  public void test1()  throws Throwable  {
      ExceptionMapper exceptionMapper0 = new ExceptionMapper();
      Class<Exception> class0 = Exception.class;
      ExceptionHandlerImpl<Exception> exceptionHandlerImpl0 = (ExceptionHandlerImpl<Exception>) mock(ExceptionHandlerImpl.class, new ViolatedAssumptionAnswer());
      doReturn((String) null).when(exceptionHandlerImpl0).toString();
      exceptionMapper0.map(class0, exceptionHandlerImpl0);
      ExceptionHandlerImpl exceptionHandlerImpl1 = exceptionMapper0.getHandler(class0);
      ;
}
 @Test(timeout = 4000)
  public void test2()  throws Throwable  {
      ExceptionMapper exceptionMapper0 = new ExceptionMapper();
      // Undeclared exception!
      try { 
        exceptionMapper0.getHandler((Exception) null);
        fail("Expecting exception: NullPointerException");
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("spark.ExceptionMapper", e);
      }
  }
 @Test(timeout = 4000)
  public void test3()  throws Throwable  {
      ExceptionMapper exceptionMapper0 = ExceptionMapper.getServletInstance();
      ;
}
 @Test(timeout = 4000)
  public void test4()  throws Throwable  {
      ExceptionMapper exceptionMapper0 = new ExceptionMapper();
      // Undeclared exception!
      try { 
        exceptionMapper0.getHandler((Class<? extends Exception>) null);
        fail("Expecting exception: NullPointerException");
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("spark.ExceptionMapper", e);
      }
  }
 @Test(timeout = 4000)
  public void test5()  throws Throwable  {
      ExceptionMapper exceptionMapper0 = ExceptionMapper.getInstance();
      MockException mockException0 = new MockException();
      Class<Exception> class0 = Exception.class;
      exceptionMapper0.getHandler(class0);
      ExceptionHandlerImpl exceptionHandlerImpl0 = exceptionMapper0.getHandler((Exception) mockException0);
      ;
}
 @Test(timeout = 4000)
  public void test6()  throws Throwable  {
      ExceptionMapper exceptionMapper0 = ExceptionMapper.getInstance();
      ExceptionMapper exceptionMapper1 = ExceptionMapper.getServletInstance();
      ;
}
 @Test(timeout = 4000)
  public void test7()  throws Throwable  {
      ExceptionMapper exceptionMapper0 = ExceptionMapper.getInstance();
      exceptionMapper0.clear();
  }
